doc/nvim-tree-lua.txt

```diff
@@ -2118,6 +2119,7 @@ You are encouraged to copy these to your own |nvim-tree.on_attach| function.
   vim.keymap.set('n', 'C',       api.tree.toggle_git_clean_filter,    opts('Toggle Filter: Git Clean'))
   vim.keymap.set('n', '[c',      api.node.navigate.git.prev,          opts('Prev Git'))
   vim.keymap.set('n', ']c',      api.node.navigate.git.next,          opts('Next Git'))
+  vim.keymap.set('n', ']g',      api.node.navigate.git.next_recurse,  opts('Next Git Recursive'))
```

Comment:
We can reuse next/prev with an opt. Something like:

```help
node.navigate.git.next({opts})        *nvim-tree-api.node.navigate.git.next()*
    Navigate to the next item showing git status.

    Parameters: ~
      • {opts} (table) optional parameters

    Options: ~
      • {recurse}     (boolean)  recurse directories containing items showing status, default false
```

Reply:
This`recurse` opt would also need to be added for diagnostics and opened then - unless a global `recurse` opt would make more sense? And apply to all three categories.

Reply:
There are too many options;  we're preferring API to options wherever possible.

Reply:
I'd be most grateful if you could add a type for all navigations:

```lua
---@class ApiNodeNavigateOpts
---@field recurse boolean|nil default false
```